### ISSUE TITLE
Unlocked Immortal Spell Effects feature have a bug on versus screen, featured hero, and overview

### ISSUE THREAD

**Issue by SkyOlredja** [jeffhill/Dota2#1138]
### Description

this feature is cool but when on the versus screen, featured hero in dota profile, and overview after the game. the items we use are replaced by immortal items that we only need the spell effect for.

### Screenshots

![1](https://user-images.githubusercontent.com/107340061/173222232-9b521542-5fca-4d39-9a60-96cb834ce12a.jpg)
![2](https://user-images.githubusercontent.com/107340061/173222234-d1bed15e-72bf-49e9-a66c-c626ec9e359e.jpg)
![3](https://user-images.githubusercontent.com/107340061/173222235-3553e67f-1da1-44d4-b439-2b47ae8519bd.jpg)
![4](https://user-images.githubusercontent.com/107340061/173222236-12385b9a-305c-4435-8646-c2589993d146.jpg)

**Comment by SnabrielSnarsch** [comment]
and also in MVP showcase

**Comment by SkyOlredja** [comment]
yes there too, I didn't screenshot it because I thought it was the same as overview

![20220613122324_1](https://user-images.githubusercontent.com/107340061/173285613-76d3870c-e510-4e8a-ab21-de8a6ecd6f9c.jpg)

**Comment by SkyOlredja** [comment]
> and also in MVP showcase

yes there too, I didn't screenshot it because I thought it was the same as overview

![20220613122324_1](https://user-images.githubusercontent.com/107340061/173286217-18abe0d6-73a3-4c39-92b1-423f12c1b745.jpg)
